### PR TITLE
feat: updates to config format for evaluation

### PR DESCRIPTION
**Pull request by bradegler**

### CODE REVIEW

File: pkg/server/config/config_test.go [hunk 1]
```diff
@@ -194,3 +195,103 @@ func TestConfigCompile(t *testing.T) {
...
+			},
+			token:     token,
+			want:      false,
+			expErr:    true,
+			expErrMsg: "failed to evaluate CEL expression: no such key: doesntexist",
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
```
Comment (yolocs) [inline pkg/server/config/config_test.go:273]: I believe starting at Go 1.22. You don't have to do this anymore. https://tip.golang.org/doc/go1.22

**Review by yolocs — APPROVED** [review]

File: pkg/server/config/config_test.go (hunk 1 above)
Comment (bradegler) [reply pkg/server/config/config_test.go:273]: ack

File: pkg/server/config/config.go [hunk 2]
```diff
@@ -57,12 +63,17 @@ func (r *Rule) compile(env *cel.Env) error {
...
+	if s.Rule != nil {
+		return s.Rule.compile(env)
+	}
+	return nil
 }
 
 func (c *Config) compile(env *cel.Env) error {
-	if err := c.Rule.compile(env); err != nil {
-		return fmt.Errorf("error compiling configuration ruleset: %w", err)
+	if c.Rule != nil {
```
[the remark this answers is not in the document]
Comment (bradegler) [reply pkg/server/config/config.go:73]: That feels bizare to me but I'll try it in the next PR.

File: pkg/server/config/config_evaluator.go [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+		return nil, fmt.Errorf("failed to create CEL environment: %w", err)
+	}
+	// create and configure all of the file loaders
+	localLoader := newCachingConfigLoader(expireAt,
+		newCompilingConfigLoader(env, &localConfigFileLoader{configDir: localConfigDir}))
+	inRepoLoader := newCachingConfigLoader(expireAt,
+		newCompilingConfigLoader(env, &ghInRepoConfigFileLoader{
+			provider: func(ctx context.Context, org, repo string) (*github.Client, error) {
+				return makeGitHubClient(ctx, app, org, repo)
+			},
```
[the remark this answers is not in the document]
Comment (bradegler) [reply pkg/server/config/config_evaluator.go:56]: I'll catch this in a follow up.

**Pull request merged**